packages/block-library/src/code/style.scss

```diff
@@ -1,5 +1,9 @@
 // Provide a minimum of overflow handling.
-.wp-block-code code {
-	white-space: pre-wrap;
-	overflow-wrap: break-word;
+.wp-block-code {
+	font-size: var(--wp--preset--font-size--extra-small, 0.9em);
```

Comment:
:wave: we shouldn't have used a CSS Custom Property here, explained the rationale, and prepared a bugfix at https://github.com/WordPress/gutenberg/pull/27672